src/models/userModel.js

```diff
@@ -42,18 +42,13 @@ const userSchema = new mongoose.Schema({
     default: 'xxxx',
   },
 });
-
 userSchema.pre('save', async function (next) {
   // If the password field has been modified, hash the password
   if (!this.isModified('password')) return next();
-
   this.password = await bcrypt.hash(this.password, 12);
   next();
 });
-
-userSchema.methods.correctPassword = async function(candidatePassword) {
-  return await bcrypt.compare(candidatePassword, this.password);
-};
-
+userSchema.methods.correctPassword = async (candidatePassword) =>
+  await bcrypt.compare(candidatePassword, this.password);
```

Comment:
don't use arrow function(=>) here (this) doesn't refer to current document    use regular function instead like this :

userSchema.methods.correctPassword = async function (candidatePassword) {
	return await bcrypt.compare(candidatePassword, this.password);
};